lib/executor/per_vu_iterations.go

```diff
@@ -151,13 +152,13 @@ var _ lib.Executor = &PerVUIterations{}
 
 // Run executes a specific number of iterations with each configured VU.
 // nolint:funlen
-func (pvi PerVUIterations) Run(ctx context.Context, out chan<- stats.SampleContainer) (err error) {
+func (pvi PerVUIterations) Run(parentCtx context.Context, out chan<- stats.SampleContainer) (err error) {
```

Comment:
Why is this a "parent" context for only a few executors and not for all? :)

Reply:
Fair enough, I'll change the rest, so we keep it consistent. I changed it here since we had local per-VU `ctx` values that shadowed the top-level one, causing a bit of confusion (and a bug that I only found out due to the reluctantly written tests :sweat_smile:). But no reason why we can't name it `parentCtx` everywhere.

Reply:
OK, sounds good. Maybe it would be better to avoid "parent"/"child" references (and plain `ctx` names in general, unless the scope is limited) and instead properly describe it: `runCtx`?

Reply:
Hmm ordinarily, I'd say yes... in this case though, naming them `runCtx` might be more misleading than not :disappointed: In essence, there is no `runCtx`, there are 2 :sweat_smile: They are the well-named run contexts we create inside of the `Run()` methods - `maxDurationCtx` and `regDurationCtx`. 

This context we pass to `Run()` is, ordinarily, not going to interrupt the executors. It basically is only going to get cancelled if something external to the execution interrupts the script - a crossed threshold with `abortOnFail: true`, the user hitting Ctrl+C, etc. So, I think I slightly prefer the `parentCtx` name, to signal its external nature, normally unrelated to the actual running of the executor.

Reply:
Resolved by https://github.com/loadimpact/k6/pull/1529/commits/e2bbb8d5784dd5bc34f20efd5c663bf20f24e17a